src/HttpCache/EventListener/PurgeListener.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\HttpCache\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\HttpCache\PurgerInterface;
+use ApiPlatform\Core\HttpCache\ResourceManagerInterface;
+use ApiPlatform\Core\Util\RequestAttributesExtractor;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+use Symfony\Component\HttpKernel\Event\PostResponseEvent;
+
+/**
+ * Purges the modified resource from the cache.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class PurgeListener
+{
+    private $resourceManager;
+    private $purger;
+    private $iriConverter;
+
+    public function __construct(ResourceManagerInterface $resourceManager, PurgerInterface $purger, IriConverterInterface $iriConverter)
+    {
+        $this->resourceManager = $resourceManager;
+        $this->purger = $purger;
+        $this->iriConverter = $iriConverter;
+    }
+
+    /**
+     * Stores the original IRI of the object to be deleted because it may not be available anymore at purge time (e.g. when using Doctrine).
+     *
+     * @param GetResponseForControllerResultEvent $event
+     */
+    public function onKernelView(GetResponseForControllerResultEvent $event)
```

Comment:
Probably useless because it is already done for Doctrine and not reliable enough for non-Doctrine.